### PR TITLE
refactor: use stricter channel types

### PR DESCRIPTION
**Pull request by domoritz**
Fixes https://github.com/vega/vega-lite/issues/5173

### CODE REVIEW

File: src/scale.ts [hunk 1]
```diff
@@ -854,32 +852,3 @@ export function channelSupportScaleType(channel: Channel, scaleType: ScaleType):
       return scaleType === 'ordinal'; // shape = lookup only
   }
 }
-
-export function getSupportedScaleType(channel: Channel, fieldDefType: Type) {
```
Comment (domoritz) [inline src/scale.ts:858]: this code is unused

File: test/scale.test.ts [hunk 2]
```diff
@@ -96,57 +88,4 @@ describe('scale', () => {
       }
     });
   });
-
```
Comment (domoritz) [inline test/scale.test.ts:99]: remove test for unused code

File: src/encoding.ts [hunk 3]
```diff
@@ -429,15 +432,16 @@ export function extractTransformsFromEncoding(oldEncoding: Encoding<any>, config
             }
           }
         }
+
         // now the field should refer to post-transformed field instead
-        encoding[channel] = newFieldDef;
+        encoding[channel as any] = newFieldDef;
```
Comment (domoritz) [inline src/encoding.ts:437]: need this to resolve: Expression produces a union type that is too complex to represent.ts(2590)

File: src/encoding.ts [hunk 4]
```diff
@@ -576,9 +582,9 @@ export function fieldDefs<F extends Field>(encoding: EncodingWithFacet<F>): Fiel
   return arr;
 }
 
-export function forEach<U extends Partial<Record<Channel, any>>>(
+export function forEach<U extends Record<any, any>>(
```
Comment (Saba9) [inline src/encoding.ts:585]: Why go from the more strict type of `Partial<Record<Channel, any>>>` to the less strict `Record<any, any>`?

Comment (domoritz) [reply src/encoding.ts:585]: I don't know exactly why but we actually get better type inference where we use `forEach`.

**Pull request merged**